shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rule/EncryptRule.java

```diff
@@ -125,24 +127,35 @@ private boolean isValidColumnConfigurationWithAlgorithmProvided(final AlgorithmP
     /**
      * Find encryptor.
      *
+     * @param schemaName schema name
      * @param logicTable logic table name
      * @param logicColumn logic column name
      * @return encryptor
      */
-    public Optional<EncryptAlgorithm> findEncryptor(final String logicTable, final String logicColumn) {
-        return tables.containsKey(logicTable) ? tables.get(logicTable).findEncryptorName(logicColumn).map(encryptors::get) : Optional.empty();
+    public Optional<EncryptAlgorithm> findEncryptor(final String schemaName, final String logicTable, final String logicColumn) {
+        if (!tables.containsKey(logicTable)) {
+            return Optional.empty();
+        }
+        Optional<EncryptAlgorithm> optionalEncryptor = tables.get(logicTable).findEncryptorName(logicColumn).map(encryptors::get);
+        if (optionalEncryptor.isPresent()) {
+            Properties props = optionalEncryptor.get().getProps();
```

Comment:
@cheese8 Maybe it can be simplified to `encryptAlgorithm.ifPresent(optional -> optional.getProps().putAll(EncryptPropertiesBuilder.getProperties(schemaName, "", logicTable, logicColumn)));`.

Reply:
ok, fixed.